core/templates/domain/statistics/ExplorationStatsObjectFactory.ts

```diff
@@ -62,10 +62,8 @@ export class ExplorationStats {
   }
 
   getStateStats(stateName: string): StateStats {
-    if (!this.stateStatsMapping.has(stateName)) {
-      throw new Error('no stats exist for state: ' + stateName);
-    }
-    return this.stateStatsMapping.get(stateName);
+    return this.stateStatsMapping.has(stateName) ?
```

Comment:
This seems iffy. I think it should be more deterministic? Otherwise it may shadow errors that arise due to missing stats in stateStatsMapping.

(I could probably be convinced otherwise, though. Maybe outline your reasoning -- keeping in mind that we want to take the stricter approach where possible?)